Features page responsive

koksyncho opened the pull request:
features.html:
-added an encapsulating div with class "features" which spans across the whole body so that we can make the difference between the index.html and the features.html in our style.css file (this is why the changes which github shows are so messy);
-split the col-8 which has the tesplan.png image in it into col-3, col-2 and col-3 classes for better styling;
-removed some of the elements from the html and added them from the css (mainly the last check-list with the display flex property); 

style.css:
-changes to the width and alignment of the different col-? classes and the img and li tags;

atodorov commented on content/pages/features.html:
```diff
@@ -5,152 +5,157 @@
-            <div class="col-12 screenshot-title">
-                <h3>A central dashboard to start your day</h3>
-            </div>
+        <div class="features">
```
You can break the indenting rules a little bit here and indent with 2 spaces from the body tag and not indent the rest of the blocks. Let me see if I can review the rest easily as-is, if not you can apply the change I'm talking about.

atodorov replied:
Hmmm, I have a better idea, why not apply the features class to the body tag instead ? If you're using it as a selector only then this will do.

atodorov commented on content/pages/features.html:
```diff
@@ -5,152 +5,157 @@
+                    
+                    <div class="col-2"></div>
+
+                    <div style="margin-bottom: 5rem;" class="col-3">
```
Compare how the current layout looks vs. what's on the live website and notice that the image goes a bit to the right on a regular desktop screen. This is because of the style nesting you have introduced. The col-3 class here means 25% of the encapsulating col-8 above (which is 66,66%). This is not 25% of the overall page width!

I think you should work with the previous styles, but to make things easier to spot you will have to apply the body/features change first so the diff is easier to review.

atodorov commented on content/pages/features.html:
```diff
@@ -5,152 +5,157 @@
+                            </li>
+                        </ul>
                     </div>
+                    
```
Line with spaces here, remove the spaces.

atodorov reviewed (changes requested):
Requested changes to the HTML code.

Responsive layout look good overall. The two bulleted lists need to have the same padding as the rest of the elements, especially on the left-hand side to create an interrupted visual line.

The action buttons need some vertical spacing between them when they appear stacked on top of each other.

Note. I don't see you hiding the empty col-2 divs, which are essentially used to center the content, on smaller screens. You can try display:none; width:0 for them and they will free up some space for you.

atodorov commented:
Also use 'git rebase' instead of merge and drop the merge commit!

koksyncho commented:
Thanks for the feedback! Dropping the merge commit now.

koksyncho commented:
Integrated most of the requested changes. The ones which haven't been done accordingly are:
-the "features" class is still in the encapsulating div, the spacing was done with 2 spaces instead of the previous 4 (had troubles with linking the class to the body);
-tried to match the layout of that one section (the one with the tesplan.png) to the live version while still using the col-separation, it is similar but not entirely;

atodorov commented on theme/static/style/style.css:
```diff
@@ -403,6 +408,33 @@ pre code {
+        padding-left: 2.5%;
+        padding-right: 2.5%;
+    }
+    .features .space-filler {
```
`.features .col-2` and get rid of space-filler.

atodorov commented on theme/static/style/style.css:
```diff
@@ -450,4 +482,44 @@ pre code {
+    .features .demo-btn {
+        margin-bottom: 2.5%;
+    }
+    .features .space-filler {
```
.features .col-2

atodorov commented on content/pages/features.html:
```diff
@@ -136,21 +140,22 @@ <h3>Integrate with your favorite bug tracker</h3>
-                    <a href="http://kiwitcms.readthedocs.io" class="action-btn accent-unimportant">Read Documentation</a>
+                <div style="text-align: center;">
+                    <a href="https://demo.kiwitcms.org" class="action-btn demo-btn accent-important">Checkout Demo</a>
+                    <a href="http://kiwitcms.readthedocs.io" class="action-btn documentation-btn accent-unimportant">Read Documentation</a>
```
documentation-btn isn't used anywhere so remove it.

atodorov commented on theme/static/style/style.css:
```diff
@@ -359,10 +359,15 @@ pre code {
 
+/* Features Page */
+
+.features .horizontally-stacked-list {
```
Simply .horizontally-stacked-list is enough and remove the comment above!

atodorov commented on theme/static/style/style.css:
```diff
@@ -403,6 +408,33 @@ pre code {
         text-align: left;
     }
 
+    /*features section*/
```
features page, not section

atodorov commented on theme/static/style/style.css:
```diff
@@ -450,4 +482,44 @@ pre code {
+        max-width: 95%;
+    }
+    .features .col-8 {
+        min-width: 100%;
```
I think for all of these properties max-width is enough.

atodorov commented on theme/static/style/style.css:
```diff
@@ -450,4 +482,44 @@ pre code {
+        display: block;
+    }
+    .features li {
+        padding-left: 2.5%;
```
IMO it is better to use 1.8rem as we do everywhere else. You can do the same for img as well, set max-width to 100% and update the border-size property to border-box to make sure the 100% contain all padding. margin and borders.

atodorov reviewed (changes requested):
Other changes requested, still there's a merge commit which you need to get rid of. Also squash all of the rest commits when you rebase.

atodorov commented:
> (had troubles with linking the class to the body)

The <body> tag inside the features.html file is not the <body> tag that appears on the rendered features page! This is simply a holder for the content block of that page (e.g. like begin-end). When the content is rendered Pelican takes the structure from base.html template and replaces with the content and other blocks. 

The last commit on the source branch introduces the ability to configure a body_class variable on the page which gets used in the template. When you rebase you can remove the wrapping div and your CSS selectors should work like they do ATM.

atodorov commented on content/pages/features.html:
```diff
@@ -20,7 +20,7 @@ <h3>A central dashboard to start your day</h3>
 
                 <div class="w50" style="float: left;">
-                    <div class="about-panel-title">For test engineers</div>
+                    <div class="about-panel-title">For test enginners</div>
```
Somehow the rebase didn't work for you but the latest source branch has this typo fixed.

atodorov commented on theme/static/style/style.css:
```diff
@@ -248,8 +248,7 @@ pre code {
 
-/* Title */
-.screenshot-title {
+.screenshot-title{
```
Don't do this, we prefer having a space before {, it's that some of the blocks are missing it, but this PR purpose is not the fix the styles we don't care about.

atodorov commented on theme/static/style/style.css:
```diff
@@ -263,7 +262,7 @@ pre code {
 }
 
-.img {
+.img{
```
don't remove space

atodorov commented on theme/static/style/style.css:
```diff
@@ -292,8 +291,7 @@ pre code {
 
-/* Image Count */
-.img-index {
+.img-index{
```
don't remove space

atodorov commented on theme/static/style/style.css:
```diff
@@ -405,6 +406,33 @@ pre code {
     }
 
+    /*features page*/
+    .w50 {
```
prefix with .features

atodorov commented on theme/static/style/style.css:
```diff
@@ -405,6 +406,33 @@ pre code {
+        padding-left: 1.8rem;
+        padding-right: 1.8rem;
+    }
+    .features .space-filler {
```
No such class used in the page so remove the style definition.

atodorov commented on theme/static/style/style.css:
```diff
@@ -452,15 +480,19 @@ pre code {
-*/
 
-.ui-loader {
-    display: none;
```
Don't remove this last definition, we still need it.

atodorov commented:
@ivo0126 please provide a review for this PR

ivo0126 commented on content/pages/features.html:
```diff
@@ -49,7 +49,7 @@ <h3>A Test Plan is a collection of Test Cases</h3>
 
             <div class="col-8">
-                <div style="float:left" class="w25">
+                <div style="float:left" class="col-3 w25">
```
Move styles to `style.css`.
same goes for lines:
`22, 31, 52, 76, 116, 138, 150, 151`.

atodorov commented on theme/static/style/style.css:
```diff
@@ -1,6 +1,6 @@
 /* Reset */
 
-*{
+* {
```
I agree with you that these changes should be in a separate commit, but let's leave it like this this time.

atodorov commented on content/pages/features.html:
```diff
@@ -71,9 +71,12 @@ <h3>A Test Plan is a collection of Test Cases</h3>
-                <div style="float: right; clear:right;margin-bottom: 5rem;">
+                <div class="col-2"></div>
+
+                <div style="margin-bottom: 5rem;" class="col-3">
```
Revert this change because it skews the layout of the image and follow @ivo0126 advice to move these style definitions as CSS classes.

atodorov commented on theme/static/style/style.css:
```diff
@@ -374,6 +372,12 @@ pre code {
 
+/* Features Page */
+
+.horizontally-stacked-list {
```
Move this block above the media selector for width 1360px.

atodorov commented on theme/static/style/style.css:
```diff
@@ -440,6 +444,34 @@ pre code {
+    .features .col-2 {
+        display: none; 
+        width: 0;
+    }
```
Looking at the above block I think most of it needs to go into the media selector for width 1360px so we can display images nicely even in landscape mode.

atodorov commented on theme/static/style/style.css:
```diff
@@ -463,8 +495,22 @@ pre code {
+    /*features page*/
+    .features .w50 {
+        min-width: 95%;
+        max-width: 95%;
```
I think something like width: 100% and padding (or margin) 1.8rem on both sides, probably combined with box-sizing: border-box will have the same resizing effect and allow all the elements to have the same alignment.

atodorov reviewed (changes requested):
To recap:
* keep the white space changes as they are, but in the future don't do this
* Revert the changes around the new col-3 class
* adjust class definitions override for tablet screens (or smaller PC screen) and work from there.

koksyncho committed:
Changes to the responsive features page

Putting all of the styling in the style.css, changing the ways of alignment
of the horizontal list, and returning some of the elements to how they were
in older iterations.

koksyncho commented:
Thanks for the reviews! I think that I followed your instructions, but upon committing the changes, not all of the requests disappeared (I may have screwed something up - I used "git reset HEAD~1" to cancel my first commit).

atodorov commented on theme/static/style/style.css:
```diff
@@ -325,43 +323,93 @@ pre code {
+    float: left;
+}
+
+.w50 {
```
Don't redefine the .w50 class here. This is causing its definition to be in 2 places. Better add something like .float-left class and apply it to the containers you want floating on the left.

atodorov commented on theme/static/style/style.css:
```diff
@@ -325,43 +323,93 @@ pre code {
+}
+
+.call-to-action-buttons {
+    text-align: center;
```
Can this text-align go into the .call-to-action-field definition? If yes then you can remove this additional class.

atodorov commented on theme/static/style/style.css:
```diff
@@ -388,15 +436,11 @@ pre code {
 
     /* Screenshots section */
-    .screenshots .col-2, .img-index {
+    .screenshots .img-index {
```
Revert this change, not related to current feature.

atodorov commented on theme/static/style/style.css:
```diff
@@ -388,15 +436,11 @@ pre code {
-    .screenshots .col-8 {
-        width: 100%;
-    }
-
```
revert this, unrelated to current feature

atodorov commented on theme/static/style/style.css:
```diff
@@ -432,6 +476,39 @@ pre code {
+        padding-left: 1.8rem;
+        padding-right: 1.8rem;
+    }
+    .w25 {
```
should be `.features .w25`

atodorov commented on theme/static/style/style.css:
```diff
@@ -432,6 +476,39 @@ pre code {
+        min-width: 100%;
+        max-width: 100%;
+    }
+    .testplan {
```
Prefix with `.features` just for consistency with the rest and for better clarity.

atodorov commented on theme/static/style/style.css:
```diff
@@ -455,8 +532,24 @@ pre code {
+        padding-right: 1.8rem;
+        box-sizing: border-box;
+    }
+    .col-8 .horizontally-stacked-list {
```
Prefix with `.features` for consistency and clarity.

atodorov commented on theme/static/style/style.css:
```diff
@@ -325,43 +323,93 @@ pre code {
+    text-align: center;
+}
+
+@media screen and (max-width: 1360px){
```
This looks like it should not be here. Probably result of conflicts.

atodorov reviewed (changes requested):
Please rebase to latest and resolve conflicts. Then fix all the requested changes.

atodorov commented on content/pages/features.html:
```diff
@@ -71,9 +71,10 @@ <h3>A Test Plan is a collection of Test Cases</h3>
 
-                <div style="float: right; clear:right;margin-bottom: 5rem;">
-                    <img src="/images/features/testplan.png" class="margin0 shadow">
+                <div class="testplan">
```
There's no definition for the `.testplan` class with the removed properties.

ivo0126 replied:
What do You mean by that?

The styles for `.testplan` are defined on line `502`, if that's what You're talking about.

Or are You saying that You want the properties `float: right` and `clear: right` to be returned, since they are no longer used?

atodorov replied:
Yes, I'm saying the float and clear properties need to come back. Without them this change breaks the existing layout when on a large screen.

koksyncho replied:
Thanks for the clarification, would not have understood it otherwise.

atodorov commented on content/pages/features.html:
```diff
@@ -144,9 +145,9 @@ <h3>Integrate with your favorite bug tracker</h3>
-                <div>
-                    <a href="https://demo.kiwitcms.org" class="action-btn accent-important">Checkout Demo</a>
+            <div class="col-8 call-to-action-field">
+                <div class="call-to-action-buttons">
```
call-to-action-buttons is not used, remove it.

atodorov reviewed (changes requested):
Please make the requested changes and also resolve the conflicts so we can proceed further.

ivo0126 commented on content/pages/features.html:
```diff
@@ -110,8 +111,8 @@ <h3>Integrate with your favorite bug tracker</h3>
             <div class="col-8">
-                <div style="float:left">
-                    <ul class="check-list" style="display: flex">
+                <div>
```
Delete this `div`, since it does nothing.

ivo0126 replied:
Delete this `div`, since it does nothing.

ivo0126 commented on theme/static/style/style.css:
```diff
@@ -325,52 +323,90 @@ pre code {
+    padding-top: 2rem;
+}
+
 @media screen and (max-width: 1360px){
```
When the window width is shrinked enough for the styles in this query to apply, the images go off-center and stick to the left side of the screen, which does not fit with the overall style of the page. This can be fixed by adding the following properties inside of the query:

```
.col-8 {
    width: 97%;
}

.margin0 {
    margin-left: 7.5%;
    width: 87%;
}
```

`.margin0` refers to the images.

Or try finding another solution if this one doesn't suit you.

ivo0126 replied:
When the window width is shrinked enough for the styles in this query to apply, the images go off-center and stick to the left side of the screen, which does not fit with the overall style of the page. This can be fixed by adding the following properties inside of the query:

```
.col-8 {
    width: 97%;
}

.margin0 {
    margin-left: 7.5%;
    width: 87%;
}
```

`.margin0` refers to the images.

Or try finding another solution if this one doesn't suit you.

ivo0126 commented on theme/static/style/style.css:
```diff
@@ -400,11 +436,15 @@ pre code {
 
     /* Screenshots section */
-    .screenshots .img-index {
+    .screenshots .col-2 .img-index {
```
There  has to be a comma between `.col-2` and `.img-index {`, so the whole thing will look like `.screenshots .col-2, .img-index {`, as demanded by @atodorov.

ivo0126 replied:
There  has to be a comma between `.col-2` and `.img-index {`, so the whole thing will look like `.screenshots .col-2, .img-index {`, as demanded by @atodorov.

atodorov commented on content/pages/features.html:
```diff
@@ -71,9 +71,10 @@ <h3>A Test Plan is a collection of Test Cases</h3>
+                <div class="testplan">
+                    <img src="/images/features/testplan.png" class="testplan-img margin0 shadow">
                 </div>
+
```
Remove new line, not needed.

atodorov commented on content/pages/features.html:
```diff
@@ -110,29 +111,28 @@ <h3>Integrate with your favorite bug tracker</h3>
+                    </li>
+                </ul>
+
+
```
Remove 1 of the 2 new lines.

atodorov commented on content/pages/features.html:
```diff
@@ -71,9 +71,10 @@ <h3>A Test Plan is a collection of Test Cases</h3>
-                <div style="float: right; clear:right;margin-bottom: 5rem;">
-                    <img src="/images/features/testplan.png" class="margin0 shadow">
+                <div class="testplan">
+                    <img src="/images/features/testplan.png" class="testplan-img margin0 shadow">
```
Remove testplan-img class. It adds a double 5rem margin at the botton. The wrapping div already provides this.

atodorov commented on theme/static/style/style.css:
```diff
@@ -325,52 +323,97 @@ pre code {
+    }
+    .features .margin0 {
+        margin-left: 7.5%;
+        width: 87%;
```
Use wid/max-width 100% and margin in rems, 1.8 is the default everywhere else I believe. You can also use box-sizing: border-box to make everything fit inside the container.

koksyncho replied:
box-sizing: border-box doesn't work for me, the images go out of the screen due to the margin, that is why I left the margin at 1.8rem and lowered the width to 95%. The same goes for the comment below this one.

atodorov commented on theme/static/style/style.css:
```diff
@@ -440,6 +487,39 @@ pre code {
+    }
+    .features img {
+        min-width: 95%;
+        max-width: 95%;
```
Again, use width 100% and add padding and border-box properties to control spacing between elements. I think it is best if you comment out all styles for responsive and start working on each element one by one until you are satisfied with it.

koksyncho replied:
box-sizing: border-box doesn't work for me, the images go out of the screen due to the margin, that is why I left the margin at 1.8rem and lowered the width to 95%.

atodorov commented on theme/static/style/style.css:
```diff
@@ -440,6 +487,39 @@ pre code {
+    .features .testplan {
+        min-width: 100%;
+        max-width: 100%;
+        margin-bottom: 5rem;
```
I think margin-bottom here is redundant, isn't it ?

koksyncho replied:
I believe it isn't redundant since we removed the margin-bottom from the now non-existent testplan-img class.

koksyncho commented:
Would you like the responsive part of the style.css to be taken out completely and committed separately or, as you said in one of the latest comments, to comment all of it and start uncommenting the elements one by one as they are remade?

atodorov commented:
@koksyncho please remove the overriding of styles for responsive completely and let's first merge the white-space changes and the removal of inline styles.

koksyncho committed:
Removed overriding responsive styles

The features page responsive styles have been removed due to them
overriding each other in improper ways. This way we can proceed with
the merge of the basic features page styles and can later move on to
rewriting the responsive stylings.

koksyncho commented:
@atodorov the responsive styles on the features page have been removed. I will start writing them one by one again in order to evade running into the old overriding problems.

atodorov commented on theme/static/style/style.css:
```diff
@@ -400,11 +426,15 @@ pre code {
+    .screenshots .col-8 {
+        width: 100%;
+    }
+
```
Move this back to the top from where you have deleted it. It controls the styling of the screenshots image in the main page and for screens under 1360px we still want to have the image span the full width. With the current change the images goes on to the left which is a regression from existing behavior.

atodorov reviewed (changes requested):
Revert the `.screenshots .col-8` change and this is ready to go.
